crate/src/lib.rs

```diff
@@ -0,0 +1,144 @@
+use std::path::{Path, PathBuf};
+use std::process::{Child, Command};
+use std::hash::{Hash, Hasher};
+
+use anyhow::anyhow;
+use binary_install::Cache;
+use siphasher::sip::SipHasher13;
+
+const fn platform() -> &'static str {
+    #[cfg(target_os = "linux")]
+    return "Linux";
+
+    #[cfg(all(target_os = "macos", target_arch = "x86_64"))]
+    return "Darwin";
+
+    #[cfg(all(not(all(target_os = "macos", target_arch = "x86_64")), not(target_os = "linux")))]
+    compile_error!("Unsupported platform");
+}
+
+fn local_addr(port: u16) -> String {
+    format!("0.0.0.0:{}", port)
+}
+
+// HACK: Taken from binary-install to get generated dir
+fn hashed_dirname(url: &str, name: &str) -> String {
+    let mut hasher = SipHasher13::new();
+    url.hash(&mut hasher);
+    let result = hasher.finish();
+    let hex = hex::encode(&[
+        (result >> 0) as u8,
+        (result >> 8) as u8,
+        (result >> 16) as u8,
+        (result >> 24) as u8,
+        (result >> 32) as u8,
+        (result >> 40) as u8,
+        (result >> 48) as u8,
+        (result >> 56) as u8,
+    ]);
+    format!("{}-{}", name, hex)
+}
+
+fn bin_url() -> String {
+    format!(
+        "https://cloudflare-ipfs.com/ipfs/QmZ6MQ9VMxBcahcmJZdfvUAbyQpjnbHa9ixbqnMTq2k8FG/{}-near-sandbox.tar.gz",
```

Comment:
```suggestion
        "https://ifps.io/ipfs/QmZ6MQ9VMxBcahcmJZdfvUAbyQpjnbHa9ixbqnMTq2k8FG/{}-near-sandbox.tar.gz",
```

Was having issues with the cloudflare. Soon to be replaced with AWS, but that's the easy part. ;-p